tests/test_appmesh/test_appmesh.py

```diff
@@ -148,3 +156,501 @@ def test_tag_and_list_tags_for_resource(client):
     tags = connection["tags"]
     assert tags[0] == {"key": "owner", "value": "moto"}
     assert tags[1] == {"key": "organization", "value": "moto"}
+
+
+@mock_aws
+def test_create_describe_list_update_delete_virtual_router(client):
+    MESH_NAME = "mock_mesh"
+    connection = client.create_mesh(
+        meshName=MESH_NAME,
+        spec={
+            "egressFilter": {"type": "DROP_ALL"},
+            "serviceDiscovery": {"ipPreference": "IPv4_ONLY"},
+        },
+        tags=[{"key": "owner", "value": "moto"}],
+    )
+    assert "mesh" in connection
+    mesh = connection["mesh"]
+    assert "metadata" in mesh
+    mesh_owner = mesh["metadata"]["meshOwner"]
+
+    ROUTER_1 = "router1"
+    ROUTER_2 = "router2"
+
+    connection = client.create_virtual_router(
+        meshName=MESH_NAME,
+        meshOwner=mesh_owner,
+        virtualRouterName=ROUTER_1,
+        spec={"listeners": [{"portMapping": {"port": 80, "protocol": "http"}}]},
+        tags=[{"key": "router_traffic", "value": "http"}],
+    )
+    router1 = connection.get("virtualRouter")
+    assert router1 is not None
+    assert router1["meshName"] == MESH_NAME
+    assert router1["metadata"]["meshOwner"] == mesh_owner
+    assert router1["metadata"]["version"] == 1
+    assert router1["spec"]["listeners"][0]["portMapping"]["port"] == 80
+    assert router1["spec"]["listeners"][0]["portMapping"]["protocol"] == "http"
+    assert router1["status"]["status"] == "ACTIVE"
+
+    connection = client.create_virtual_router(
+        meshName=MESH_NAME,
+        meshOwner=mesh_owner,
+        virtualRouterName=ROUTER_2,
+        spec={"listeners": [{"portMapping": {"port": 443, "protocol": "http2"}}]},
+        tags=[{"key": "router_traffic", "value": "https"}],
+    )
+    router2 = connection.get("virtualRouter")
+    assert router2 is not None
```

Comment:
This assertion is technically not necessary, just because the next line immediately uses the variable. If the variable is None, the test will fail anyway with a `KeyError`. There are quite a few examples like this.

For me, it would be cleaner to remove it - just because less lines of code make it easier to read.

It's very much personal preference though, and not particularly important, so I'm not expecting any changes. It's just an FYI - if you find it easier to read _with_ the explicit assertions, feel free to keep it. :slightly_smiling_face: